e2e/tests/vscode.test.ts

```diff
@@ -13,54 +13,80 @@ test.describe('VSCode Tests', () => {
-  });
-
   test('Should open Extensions tab and verify installed extension', async () => {
     const window = vscodeApp.getWindow();
     await window.waitForTimeout(5000);
-    const kaiTab = await window.getByRole('tab', { name: 'Konveyor' });
-    await kaiTab.click();
-    await window.waitForTimeout(500);
-    const heading = await window.getByRole('heading', {
+    await vscodeApp.openLeftBarElement(LeftBarItems.Konveyor);
+    const heading = window.getByRole('heading', {
       name: 'Konveyor',
       exact: true,
     });
     await expect(heading).toBeVisible();
-    await window.screenshot({ path: 'kai-installed-screenshot.png' });
+    await window.screenshot({
+      path: './screenshots/kai-installed-screenshot.png',
+    });
   });
 
-  test('Set Up Konevyor and Start analyzer', async () => {
+  test('Set Sources and targets', async () => {
+    await vscodeApp.getWindow().waitForTimeout(5000);
+    await vscodeApp.selectSourcesAndTargets([], ['quarkus']);
+  });
+
+  test('Set Up Konveyor and Start analyzer', async () => {
     const window = vscodeApp.getWindow();
     await window.waitForTimeout(5000);
     await vscodeApp.openSetUpKonveyor();
-    await window.waitForSelector('h3.step-title:has-text("Start Server")');
-    await window.locator('h3.step-title:has-text("Start Server")').click();
-    await window.waitForSelector('span:text("Start Analyzer")');
-    await window.locator('span:has-text("Start Analyzer")').click();
-    const konveyorView = await window.getByText('Konveyor Analysis View');
-    await expect(konveyorView).toBeVisible();
-    // Below code can be Uncommented after setting model provider
-    // const serverStatusframe = await vscodeApp.getServerStatusIframe();
-    // if (serverStatusframe) {
-    //   const serverRunning = serverStatusframe.locator(
-    //     'span.pf-v6-c-label__text',
-    //     { hasText: 'Running' }
-    //   );
-    //   console.log('Waiting for Server to start...');
+    await window.waitForTimeout(5000);
+    await window
+      .getByRole('button', { name: 'Configure Generative AI' })
+      .click();
+    await window.waitForTimeout(5000);
+    await window
+      .getByRole('button', { name: 'Configure GenAI model settings file' })
+      .click();
+    await window.waitForTimeout(5000);
+
+    await window.keyboard.press('Control+a+Delete');
+    await vscodeApp.pasteContent(
+      [
+        'models:',
+        '  OpenAI: &active',
+        '    environment:',
+        `      OPENAI_API_KEY: "${process.env.OPENAI_API_KEY}"`,
+        '    provider: "ChatOpenAI"',
+        '    args:',
+        '      model: "gpt-4o"',
+        'active: *active',
+      ].join('\n')
+    );
+    await window.keyboard.press('Control+s');
 
-    //   await serverRunning.waitFor();
-    //   console.log('Server is Running');
-    // }
+    await window.waitForTimeout(5000);
+    await vscodeApp.openSetUpKonveyor();
+    await window.waitForTimeout(5000);
```

Comment:
These waits are also replaced by waitforselector.
You need to rebase off main branch ? 
https://github.com/konveyor/kai-ci/blob/main/e2e/tests/vscode.test.ts#L36